docs/ADR-49-signed-fetch-v2.md

```diff
@@ -0,0 +1,365 @@
+# Signed Fetch V2
+
+This document describe an improved mechanism to communicate Users with Decentraland Services over HTTP messages. This mechanism support all previously use case and includes new security features and follow common and well proved standards:
+
+## Summary
+
+Preserved features include:
+
+- The ability to generate signed request on front end application
+- Sign request's method and pathname
+- Sign metadata
+- Support `decentraland-crypto` as signature generator
+
+New features include:
+
+- Sign `Host` and `Content-Type` headers and query to prevent request re-utilization between services and/or environments
+- Sign arbitrary headers that Services can trust that were provide by Users
+- Sign request's body to prevent manipulation between Users and Services
+- Use `Authorization` header to take advantage of default behavior of caching for private content (see: [1](https://greenbytes.de/tech/webdav/rfc7234.html#rfc.section.3), [2](https://www.w3.org/Protocols/HTTP/Issues/cache-authentication.html))
+- Support `multipart/form-data` requests
+- Support `PersonalSign` as signature generator
+- Self contain expiration
+
+> Although this new mechanism is not backward compatible it doesn't include any incompatibility which means tha a Service could accept request signed with either of the two mechanisms
+
+## Sign a Request
+
+In this mechanism the `Authorization` header is used to transport the signature of the request:
+
+### Authorization
+
+The [Authorization](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#authorization_and_proxy-authorization_headers) request headers contain the credentials to authenticate a User with a Service. Here, the `Type` is used to differentiate the format on which the `Credentials` is encode/encrypted.
+
+```ts
+  Authorization = Type + ' ' + Credentials
+```
+
+In mechanism `Type` is compose of 3 component:
+
+```ts
+  Type = SignAlgorithm + '+' + HashAlgorithm [ + '+' + SignEncoding ]
+```
+
+- `SignAlgorithm`: Defines how user generate the signature of the request
+  - `DCL`: when the signature was generated using [`decentraland-crypto`](https://github.com/decentraland/decentraland-crypto)
+  - `SIGN`: when the signature was generated using a [`eth_sign`](https://eth.wiki/json-rpc/API#eth_sign) call
+
+> Note: Other methods can be added into this list, like `DCL2` for future versions of `decentraland-crypto` or the algorithm change and `TYPED4` if `signTypedData_v4` is use to generate the signature
+
+- `HashAlgorithm`: Defines the Hash Algorithm used to hash the request, since [some algorithms as not consider secure](https://en.wikipedia.org/wiki/Secure_Hash_Algorithms) any more, `SHA256` is support as minimum
```

Comment:
```suggestion
- `HashAlgorithm`: Defines the algorithm used to hash the request, since [some algorithms as not consider secure](https://en.wikipedia.org/wiki/Secure_Hash_Algorithms) any more, `SHA256` is support as minimum
```